aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
+
+
+array : '[' identifier_list ']'
+
+assignment_statement: identifier type '=' expression
+                    | '!' assignment_statement
```

Comment:
Not sure this is the best way to define this. Syntactically it makes sense, but semantically it doesn't. The `!` operator on identifier declarations marks the variable as mutable. That should probably be grouped into something like an `identifier_decl` rule that is then substituted into L15.